trino/dbapi.py

```diff
@@ -202,7 +202,18 @@ def __init__(
         else:
             self._http_session = http_session
         self.http_headers = http_headers
-        self.http_scheme = http_scheme if not parsed_host.scheme else parsed_host.scheme
+
+        # Set http_scheme
+        if parsed_host.scheme:
+            self.http_scheme = parsed_host.scheme
+        elif http_scheme:
+            self.http_scheme = http_scheme
+        elif port == constants.DEFAULT_TLS_PORT:
+            self.http_scheme = constants.HTTPS
+        elif port == constants.DEFAULT_PORT:
+            self.http_scheme = constants.HTTP
+        else:
+            self.http_scheme = constants.HTTP
```

Comment:
this would be when there is no scheme in the URI + non-standard port is used like `coordinator:8090`, right?

Reply:
yes, + `http_scheme` isn't explicitly set